### PR TITLE
Fix docker not running not bubbling up correctly in run config

### PR DESCRIPTION
**Pull request by abrooksv**
We need to let the error handling fire when the success callback throws an error in the StepExecutor

## Related Issue(s)
#2418

## Screenshots (if appropriate)
Screen Shot 2021-02-15 at 2 23 45 PM

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunner.kt [hunk 1]
```diff
@@ -56,10 +57,10 @@ open class SamRunner(protected val settings: LocalLambdaRunSettings) {
             processHandler.waitFor()
             val exitValue = processHandler.exitCode
             if (exitValue != 0) {
-                throw Exception(message("lambda.debug.docker.not_connected"))
+                throw ExecutionException(message("lambda.debug.docker.not_connected"))
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunner.kt:60]: can we write-out stderr to help the user out a bit? (so they can tell if docker was not in the PATH, docker internal failure, etc.)

**Review by rli — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunner.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunner.kt:60]: Hmmm... We should probably move this to a Step so it has a console to write to?

File: .changes/next-release/bugfix-8ec0ed8f-80ee-40c2-814d-f474eeee55d2.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "type" : "bugfix",
+  "description" : "Fix uncaught error when local Lambda fails due to docker is not running"
```
Comment (hunterwerlla) [inline .changes/next-release/bugfix-8ec0ed8f-80ee-40c2-814d-f474eeee55d2.json:3]: Add the issue number in the changelog too

**Review by hunterwerlla — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunner.kt (hunk 1 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunner.kt:60]: ya, can be a follow-up later

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunner.kt:60]: Will convert this to a step after #2436 is merged

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunner.kt:60]: Converted

**Comment by abrooksv** [comment]
Screen Shot 2021-02-17 at 5 16 56 PM

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInvokeRunner.kt [hunk 3]
```diff
@@ -229,7 +227,7 @@ class SamInvokeRunner : AsyncProgramRunner<RunnerSettings>() {
     private fun buildWorkflow(environment: ExecutionEnvironment, buildRequest: BuildRequest, samOptions: SamOptions): StepExecutor {
         val buildStep = BuildLambda(buildRequest.template, buildRequest.logicalId, buildRequest.buildDir, buildRequest.buildEnvVars, samOptions)
 
-        return StepExecutor(environment.project, message("sam.build.running"), StepWorkflow(buildStep), environment.executionId.toString())
+        return StepExecutor(environment.project, message("sam.build.running"), StepWorkflow(CheckDockerStep(), buildStep), environment.executionId.toString())
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInvokeRunner.kt:230]: Don;t we need this step on deploy `createLambdaWorkflowForImage/zip(` and update?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInvokeRunner.kt:230]: We dont have it today, but we should add it.

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInvokeRunner.kt]: Done

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**